pallets/subtensor/src/lib.rs

```diff
@@ -1105,6 +1108,21 @@ pub mod pallet {
 			Self::do_registration(origin, netuid, block_number, nonce, work, hotkey, coldkey)
 		}
 
+		/// Associates the hotkey to this coldkey.
+		/// TODO(const): this call should be costly.
+		#[pallet::call_index(51)]
+
+		/* Instead of using high arbitrary weight, we need to use inclusion fees. See [here](https://docs.substrate.io/build/tx-weights-fees/#custom-fees)*/
+		#[pallet::weight((Weight::from_ref_time(2_000_000_000)
+		.saturating_add(T::DbWeight::get().reads(27))
+		.saturating_add(T::DbWeight::get().writes(22)), DispatchClass::Normal, Pays::No))]
+		pub fn associate( 
```

Comment:
How were these weights calculated? Was it benchmarked on one of the validator machines?

Reply:
I believe the weight was arbitrarily picked by @unconst but i’ll be benchmarking both associate and de associate calls and fix the weights.